HW_4/HW04/HW04.Task5/Program.cs

```diff
@@ -0,0 +1,160 @@
+
+            for (int i = 1; i < strArray.Length; i++)
+            {
+                if (strArray[i].Length > longestWord.Length)
+                {
+                    longestWord = strArray[i];
+                }
+                if (strArray[i].Length < shortestWord.Length)
+                {
+                    shortestWord = strArray[i];
+                }
+            }
+            int indexLong = Array.IndexOf(strArray, longestWord);
+            int indexShort = Array.IndexOf(strArray, shortestWord);
+
+            strArray[indexLong] = shortestWord;
+            strArray[indexShort] = longestWord;
+
+            str2 = string.Join(" ", strArray);
+
+            return str2;
+        }
+
+        private static int CountAllSymbols(string str)
+        {
+            str = str.Replace(" ", "");
+
+            int symbolQuantity = 0;
+
+            for (int i = 0; i < str.Length; i++)
+            {
+                // checking for Russian letters and all punctuation marks
+                if (!Regex.IsMatch(str[i].ToString(), @"^[а-яёА-Я]+\p{P}+$"))
+                {
+                    symbolQuantity += 1;
+                }
+            }
+
+            return symbolQuantity;
+        }
+
+        private static int FindLongest(string[] strArray)
+        {
+            string longest = strArray[0];
+            int longestIndex = 0;
+            for (int i = 1; i < strArray.Length; i++)
+            {
+                if (strArray[i].Length > longest.Length)
+                {
+                    longest = strArray[i];
+                    longestIndex = i;
+                }
+            }
+            return longestIndex;
+        }
+
+        private static string[] SelectionSort(string[] strArray)
+        {
+            string[] newArray = strArray;
+
+            string[] newArr = new string[] { };
+            for (int i = 0; i < strArray.Length; i++)
+            {
+                int longestIndex = FindLongest(strArray);
+                newArray[i] = strArray[longestIndex];
+
+                Array.Clear(strArray, longestIndex, 1);
+            }
+            return newArray;
+        }
+
+        private static string[] SortString(string str)
+        {
+            string[] strArray = str.Split(new char[] { ' ' });
+
+            string[] newArray = SelectionSort(strArray);
+
+            return newArray;
+        }
+
+        static void Main(string[] args)
+        {
+            Console.WriteLine("Please, input several words:");
+            string wordsLine = Console.ReadLine();
+
+            //string str = SwapWords(wordsLine);
+
+            //int symbolQuantity = CountAllSymbols(wordsLine);
+            //Console.WriteLine(symbolQuantity);
+
+            string[] newArray = SortString(wordsLine);
+
+            foreach (var item in newArray)
+            {
+                Console.WriteLine(item);
+            }
+
+            Console.ReadKey();
+        }
+    }
+}
```

Comment:
better -put all methods under Main, 

for instance, in the following order

Main
SortString
SelectionSort
FindLongest